### PR TITLE
✨ Detect which inspection implementation to use for new nodes

### PR DESCRIPTION
**Pull request by dtantsur**

### CODE REVIEW

File: pkg/provisioner/ironic/validatemanagementaccess_test.go [hunk 1]
```diff
@@ -107,6 +107,44 @@ func TestValidateManagementAccessCreateNodeNoImage(t *testing.T) {
 	assert.Equal(t, createdNode.DeployInterface, "")
 }
 
+func TestValidateManagementAccessCreateNodeNewInspection(t *testing.T) {
```
Comment (kashifest) [inline pkg/provisioner/ironic/validatemanagementaccess_test.go:110]: Do we need a testcase to check if the backward compatibility is working with `inspector`  interface as well? Or is it already tested in other cases? Sorry I am a bit unfamiliar with unit tests in this part of the code

**Review by kashifest — COMMENTED** [review]
/approve

**Review by iurygregory — COMMENTED** [review]
/lgtm

**Comment by iurygregory** [comment]
LGTM  - sorry not enough power =)

**Comment by dtantsur** [comment]
/test-centos-e2e-integration-main 

IPA downloader timed out.

**Pull request merged**